### ISSUE TITLE
TreeExplainer on binary LightGBM model produces shap values for multi-class

### ISSUE THREAD

**Issue by eghamtech** [shap/shap#837]
I have come across a number of models on different data sets whereby LightGBM model clearly trained on binary data and configured to produce just a single output value between 0 and 1, however, when explained using SHAP TreeExplainer it produces SHAP values as if it is a multi-output model. When trying to do summary plot with type='dot' it then produces error:

AssertionError: Only plot_type = 'bar' is supported for multi-output explanations!

So in such cases I can only produce a bar chart but it makes little sense since the data doesn't have multiple classes and yet, in some cases TreeExplainer thinks data has 6 classes, in some cases 2 classes, which is definitely not true.

This doesn't happen always, so on most other data I worked with using same code everything works fine, shap values and dot/violin plots produced as expected, but occasionally I get data where this fails as described and I cannot see what it is so special about such data.

Are you able to assist with what may be happening here?

**Comment by candalfigomoro** [comment]
Could this be related to issue #796?

**Comment by eghamtech** [comment]
yes, it seems to be the same issue, thank you, although I don't see what the solution is, can you please advice?

**Comment by eghamtech** [comment]
Apologies, I got it. 
For those who didn't the solution is to specify which class (even for binary) i.e., 0 or 1 to extract from shap_values list to be shown on the plot since the format of the list has changed and I must have had different versions of SHAP library on different experiments:

shap.summary_plot(shap_values[0], x_test, plot_type='dot')
or 
shap.summary_plot(shap_values[1], x_test, plot_type='dot')
instead of just
shap.summary_plot(shap_values, x_test, plot_type='dot')
which worked in previous versions of SHAP
 
The only thing that is still unclear is how shap_values list may now contain predicted labels other than just 0 and 1 (in some of my data I see 6 classes i.e., 6 arrays of shap_values) whereas LightGBM output is clearly between 0 and 1 and trained with ['objective']  = 'binary'

**Comment by slundberg** [comment]
@eghamtech this was a change that was meant to create more consistency, where binary classification always produces two outputs (positive and negative classes). But if you see 6 classes please share a notebook example and we can look into it.

**Comment by eghamtech** [comment]
@slundberg thanks, it happens only on specific data sets - I'll need to think how to demonstrate it on some data that can be made available publicly.

Another question, in binary case, assuming we do get 2 sets of shap values, which one actually corresponds to the output of label 1? Is it shap_values[0] or shap_values[1]?

The reason for asking is that a reasonable assumption would be use shap_values[1] for explaining probability output of a binary classifier and shap_values[0] would be its negative, however, on the data I have it seems that it is other way around. I am not 100% sure yet, but that's what it seems like, so can you please confirm which set is supposed to explain which label in the binary case?

**Comment by VitoDH** [comment]
+1 for eghamtech. Could you please clarify which part of shap values in the returned list should I pick in the binary case? Thank you!

**Comment by eghamtech** [comment]
Unfortunately, I still don't know the definitive answer, more often than not when I look at my data results shap_values[0] seems to be giving more logical explanation of binary outcomes compared to its negative shap_values[1], but I can't be certain, so it would be good to have an answer from @slundberg

**Comment by VitoDH** [comment]
@eghamtech Me neither. Thanks for your reply! :-)

**Comment by JackPairedUp** [comment]
@eghamtech @VitoDH Have the same issue as you guys, I tested it out by including the target variable as a feature in the model.

shap.summary_plot(shap_values[1], X_train)

This returns the correct result according to this plot (sorry, had to blank out all feature names).

![image](https://user-images.githubusercontent.com/62287655/78099517-d7bc1780-742d-11ea-9903-690047d341e4.png)
